### PR TITLE
More support for cancellation tokens

### PR DESCRIPTION
**Pull request by matt-richardson**
We need more cancellation token support in our E2E tests, mainly around tasks.
This adds it in.

### CODE REVIEW

**Review by rosslovas — CHANGES_REQUESTED** [review]
Nice one, always happy to see more methods taking cancellation tokens in here 🎉 

However, I don't think this should merge without first addressing the possibility that we could cause a compile error for someone out there, despite how unlikely it might be.

There are also some failing tests, but I think they're just mocking the wrong `Get` method so should be an easy fix.

File: source/Octopus.Server.Client/OctopusAsyncClient.cs [hunk 1]
```diff
@@ -577,7 +577,7 @@ public Uri QualifyUri(string path, object parameters = null)
             return serverEndpoint.OctopusServer.Resolve(path);
         }
 
-        protected virtual async Task<OctopusResponse<TResponseResource>> DispatchRequest<TResponseResource>(OctopusRequest request, bool readResponse, CancellationToken cancellationToken = default)
+        protected virtual async Task<OctopusResponse<TResponseResource>> DispatchRequest<TResponseResource>(OctopusRequest request, bool readResponse, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (matt-richardson) [reply source/Octopus.Server.Client/OctopusAsyncClient.cs:585]: I've added the overload 👍

**Review by rosslovas — APPROVED** [review]
Nice, LGTM!

**Comment by matt-richardson** [comment]
[sc-51802]

**Pull request merged**